web_monitoring/html_diff_render.py

```diff
@@ -810,17 +926,51 @@ def html(self):
 # I had some weird concern that I needed to make this token a single word with
 # no spaces, but now that I know this differ more deeply, this is pointless.
 class ImgTagToken(tag_token):
-    def __new__(cls, tag, data, html_repr, pre_tags=None,
+
+    regex = re.compile(r"(src|srcset)=('|\")(.*?)('|\")")
+
+    def are_any_urls_equal(self, other):
+        strict_urls = self.strict_urls
+        self_array = self[7:-3]
+        self_array = self_array.split(',')
+        other_array = other[7:-3]
+        other_array = other_array.split(',')
+        for url_a in self_array:
+            for url_b in other_array:
+                if strict_urls:
+                    return StrictUrlRule.clean_resource_url(url_a[1:-1], url_b[1:-1], strict_urls)
+                elif url_a == url_b:
+                    return True
+        return False
+
+    def __new__(cls, tag, data, html_repr, strict_urls, pre_tags=None,
                 post_tags=None, trailing_whitespace=""):
-        obj = DiffToken.__new__(cls, "\n\nImg:%s\n\n" % data,
+        found_urls = ImgTagToken.regex.findall(html_repr)
```

Comment:
In `flatten_el`, we grab the actual `src` attribute value out of the `img` element: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/e602b3f94cdcbebc54db0861b38aa6f2a665b9bc/web_monitoring/html_diff_render.py#L693-L694

And then in `fixup_chunks`, we create a `tag_token` instance` where the `data` is just the `src` attribute: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/e602b3f94cdcbebc54db0861b38aa6f2a665b9bc/web_monitoring/html_diff_render.py#L636-L641

(So at this point, we’ve already parsed out the src and have it available.)

Finally, in `_customize_token`, we replace the `tag_token` instance with a `ImgTagToken`: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/e602b3f94cdcbebc54db0861b38aa6f2a665b9bc/web_monitoring/html_diff_render.py#L1036-L1044

A few things are important to note here:

- We already have the value of the `src` element in the form of `self.data`, so there’s no need to parse it out again here. We should just compare `self.data` to `other.data` when we compare.

- When parsing in `flatten_el`, we already have meaningful data in the form of `etree`, so we generally try and pull out all the important stuff there instead of doing more string parsing. At current, we get the `src` that way, but we could also get `srcset` if we want to use it in comparisons. That way we don’t have to parse it again here.

- We only do this messy middle step with `tag_token` because of the incremental way this was forked from lxml. Ideally, we’d refactor to get rid of `_customize_token` entirely. If you want to extract `srcset` for use now, you should…

    1. Modify `flatten_el` to pull that info out.
    2. Modify `fixup_chunks` to create a `ImgTagToken` instance with that data instead of making a `tag_token`.
    3. Then you should get rid of the logic in `_customize_token` that makes `ImgTagToken` instances, because it would never run anymore.
    4. Clean up the logic here and in the `are_any_urls_equal` method to use that data instead of doing more string parsing.